src/main/java/de/btegermany/terraplusminus/commands/TpllCommand.java

```diff
@@ -2,242 +2,201 @@
 
 import com.google.common.io.ByteArrayDataOutput;
 import com.google.common.io.ByteStreams;
+import com.mojang.brigadier.Command;
+import com.mojang.brigadier.arguments.StringArgumentType;
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.tree.LiteralCommandNode;
 import de.btegermany.terraplusminus.Terraplusminus;
-import de.btegermany.terraplusminus.data.TerraConnector;
 import de.btegermany.terraplusminus.gen.RealWorldGenerator;
 import de.btegermany.terraplusminus.utils.ConfigurationHelper;
 import de.btegermany.terraplusminus.utils.LinkedWorld;
-import io.papermc.lib.PaperLib;
-import io.papermc.paper.command.brigadier.BasicCommand;
 import io.papermc.paper.command.brigadier.CommandSourceStack;
+import io.papermc.paper.command.brigadier.Commands;
+import io.papermc.paper.command.brigadier.argument.ArgumentTypes;
+import io.papermc.paper.command.brigadier.argument.resolvers.selector.PlayerSelectorArgumentResolver;
 import net.buildtheearth.terraminusminus.generator.EarthGeneratorSettings;
 import net.buildtheearth.terraminusminus.projection.GeographicProjection;
 import net.buildtheearth.terraminusminus.projection.OutOfProjectionBoundsException;
+import net.buildtheearth.terraminusminus.util.geo.CoordinateParseUtils;
+import net.buildtheearth.terraminusminus.util.geo.LatLng;
 import org.bukkit.Bukkit;
+import org.bukkit.HeightMap;
 import org.bukkit.Location;
-import org.bukkit.Material;
 import org.bukkit.World;
+import org.bukkit.command.CommandSender;
+import org.bukkit.configuration.file.FileConfiguration;
 import org.bukkit.entity.Player;
+import org.bukkit.event.player.PlayerTeleportEvent;
 import org.bukkit.generator.ChunkGenerator;
 import org.jetbrains.annotations.NotNull;
 
-import static org.bukkit.ChatColor.RED;
-
-
-public class TpllCommand implements BasicCommand {
-
-    @Override
-    public void execute(@NotNull CommandSourceStack stack, @NotNull String[] args) {
-
-        Player player = (Player) stack.getSender();
-        if (!player.hasPermission("t+-.tpll")) {
-            player.sendMessage(Terraplusminus.config.getString("prefix") + "§7No permission for /tpll");
+import java.util.List;
+import java.util.Random;
+
+public class TpllCommand {
```

Comment:
This class uses experimental APIs all over the place. Imo a @SupressWarning on the class explaining why this is fine and how it will be fixed would be nice.

This adds plenty of warning noise.